tests/Common/TriggersTest.php

```diff
@@ -0,0 +1,171 @@
+    }
+
+    /**
+     * @dataProvider deleteKeyProvider
+     */
+    public function testMissingParameters($keyToDelete): void
+    {
+        $data = [
+            'component' => 'orchestrator',
+            'configurationId' => 123,
+            'coolDownPeriod' => 10,
+            'runWithTokenId' => 'nothing-is-here',
+            'tableIds' => ['nothing-is-here'],
+        ];
+        unset($data[$keyToDelete]);
+        $this->expectExceptionMessage(sprintf('Missing required query parametr(s) [%s]', $keyToDelete));
+        $this->expectException(ClientException::class);
+        $this->_client->createTrigger($data);
+    }
+
+    public function deleteKeyProvider(): array
+    {
+        return [
+            ['component'],
+            ['configurationId'],
+            ['coolDownPeriod'],
+            ['runWithTokenId'],
+            ['tableIds'],
+        ];
+    }
+
+    public function testDeleteTrigger(): void
+    {
+        $table = $this->createTableWithRandomData("watched-2");
+        $newTokenId = $this->_client->createToken([$this->getTestBucketId() => 'read']);
+
+        $trigger = $this->_client->createTrigger([
+            'component' => 'orchestrator',
+            'configurationId' => 123,
+            'coolDownPeriod' => 10,
+            'runWithTokenId' => $newTokenId,
+            'tableIds' => [
+                $table,
+            ],
+        ]);
+
+        $loadedTrigger = $this->_client->getTrigger($trigger['id']);
+        $this->assertEquals($trigger['id'], $loadedTrigger['id']);
+
+        $this->_client->deleteTrigger($loadedTrigger['id']);
+
+        $this->expectException(ClientException::class);
+        $this->expectExceptionMessage(sprintf('Trigger with id [%d] was not found', $loadedTrigger['id']));
+        $this->_client->getTrigger($trigger['id']);
+    }
+
+    public function testListAction(): void
+    {
+        $table = $this->createTableWithRandomData("watched-2");
+        $newTokenId = $this->_client->createToken([$this->getTestBucketId() => 'read']);
+
+        $trigger1ConfigurationId = time();
+        $componentName = uniqid('test');
+        $trigger1 = $this->_client->createTrigger([
+            'component' => $componentName,
+            'configurationId' => $trigger1ConfigurationId,
+            'coolDownPeriod' => 10,
+            'runWithTokenId' => $newTokenId,
+            'tableIds' => [
+                $table,
+            ],
+        ]);
+        $trigger2 = $this->_client->createTrigger([
+            'component' => 'keboola.ex-manzelka',
+            'configurationId' => 123,
+            'coolDownPeriod' => 10,
+            'runWithTokenId' => $newTokenId,
+            'tableIds' => [
+                $table,
+            ],
+        ]);
+
+        $triggers = $this->_client->listTriggers();
+        $trigger1Found = $trigger2Found = false;
+        foreach ($triggers as $trigger) {
```

Comment:
Nevím jestli neexistuje nějaký lepší způsob jak ověřit listování bez parametrů. Ono to totiž vrací všechny triggery projektu, kterých už mám XX. Mazat je na začátku mi příjde jako blbost, protože by ti to pak smazalo všechny triggery z projektu, které sis tam třeba ručně udělal.

Reply:
Všechno smazat na začátku každého testu. stejně jako se mažou tabulky, workspace, konfigurace atd.

Reply:
Stejně by se to mělo dělat jen nad test projektem, takže to smaže jen ty v tom testovacím, ne?

Reply:
jj, nad produkčním to nechceš pouštět https://github.com/keboola/storage-api-php-client#tests

Reply:
v manage api klientovi je radši ochrana aby si to oproti prod stackům vůbec  nepouštěl https://github.com/keboola/kbc-manage-api-php-client/blob/ffdfebdb55101636339d1c8ec31c0dcd4bbb3b07/tests/ClientTestCase.php#L46

Reply:
Ok všechno smažu v setUpu